Fix #25: Replace legend with togglable header.

torfsen opened the pull request:
Here's a prototype of how we could replace the legend.

My idea was that the title (including the dropdown) and the time selection controls are so important that we want to always show them. However, a legend inside the map takes up a lot of unnecessary space, especially on small screens. I've therefore moved everything into a header which can be expanded using a button.

Right now the additional information shown in the expanded header is just what we had in the legend. See #86 for a discussion of what other stuff we could put there.

There was also the suggestion by @johnjohndoe of moving the detailed content to a separate page. I don't think that a separate page is a good place for information tied to the map's state (e.g. information about the current city's data), but it could work for the general project information. Nevertheless I'd prefer to keep things in one place.

I'd love to get feedback on the prototype. Especially:

* Does the layout work on your favourite browsers/devices/screen sizes?
* Do you agree with which information is shown by default? For example, license information is currently only available once the header has been expanded.

torfsen commented:
Any thoughts, @johnjohndoe & @dirkschumacher?

johnjohndoe commented on js/main.js:
```diff
@@ -415,6 +415,48 @@ function loadCities() {
+    console.log('  header.height = ' + $('#header').outerHeight(true));
+    $('#map').outerHeight($(window).height() - $('#header').outerHeight(true));
+    console.log('  new height = ' + $('#map').outerHeight());
+}
```
You might want to remove or comment out debug output here.

torfsen replied:
Fixed.

johnjohndoe commented:
Looks good on Google Chrome developer tool preview for mobile devices:

![wim](https://cloud.githubusercontent.com/assets/144518/15294555/009be994-1b8d-11e6-8a8b-841e96cd7fdf.png)

torfsen commented:
I've removed the debug logs and cleaned up the datasource titles: Some of them simply stated the source (e.g. "Stadt Berlin") while some also had prefixes like "Daten von", e.g. "Daten der Stadt Bochum". Now they all only contain the name of the source plus additional information like licensing and date of last update.

johnjohndoe commented:
I rebased your branch and changed the titles for the cities which were added since your last rebase. Then I merged the branch. Thank you for the work!
